### PR TITLE
[Bonus] Get a ride of some fake MailQueues

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java [hunk 1]
```diff
@@ -104,6 +104,14 @@ public MailQueueItem deQueue() throws MailQueueException, InterruptedException {
             }
         }
 
+        public Mail getLastMail() throws MailQueueException, InterruptedException {
```
Comment (rouazana) [inline server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java]: no test?

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[a89b1a37d11330ce7802c1c50e105ff03e9345a4] Failed tests: 
[a89b1a37d11330ce7802c1c50e105ff03e9345a4]   DelayedManageableMailQueueContract.flushShouldPreserveBrowseOrder:76 
[a89b1a37d11330ce7802c1c50e105ff03e9345a4] Actual and expected have the same elements but not in the same order, at index 0 actual element was:
[a89b1a37d11330ce7802c1c50e105ff03e9345a4]   <"name3">
[a89b1a37d11330ce7802c1c50e105ff03e9345a4] whereas expected element was:
[a89b1a37d11330ce7802c1c50e105ff03e9345a4]   <"name1">
```

File: server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java [hunk 2]
```diff
@@ -80,7 +81,7 @@ public void enQueue(Mail mail, long delay, TimeUnit unit) throws MailQueueExcept
         @Override
         public void enQueue(Mail mail) throws MailQueueException {
             try {
-                mailItems.addFirst(new MemoryMailQueueItem(cloneMail(mail)));
+                mailItems.addLast(new MemoryMailQueueItem(cloneMail(mail)));
```
Comment (mbaechler) [inline server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java]: does that mean our contract are not good enough ?

Comment (chibenwa) [reply server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java]: No, I just switch the order we use in the queue to match the semantic `getLast` and avoid a getLast calling getFirst what would have been extremly *confusing*.

**Comment by chibenwa** [comment]
test this please (heap)

**Comment by chibenwa** [comment]
This PR is now part of https://github.com/linagora/james-project/pull/1238

**Pull request closed without merging**